modules/stanford_earth_capx/src/EarthCapxInfo.php

```
@@ -0,0 +1,249 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: kennethsharp1
+ * Date: 5/30/18
+ * Time: 9:29 PM
+ */
+
+namespace Drupal\stanford_earth_capx;
+
+use Drupal\Core\Database;
+use Drupal\migrate\MigrateMessage;
+
+/*
+ * Encapsulates an information table for CAP-X Profile imports
+ */
+class EarthCapxInfo
+{
+
+  const EARTH_CAPX_INFO_INVALID = 0;
+  const EARTH_CAPX_INFO_NEW = 1;
+  const EARTH_CAPX_INFO_FOUND = 2;
+
+  const EARTH_CAPX_INFO_TABLE = 'migrate_info_earth_capx_importer';
+
+  private $sunetid;
+  private $etag;
+  private $profilePhotoTimestamp;
+  private $entity_id;
+  private $profilePhotoFid;
+  private $status;
+
+  /**
+   * construct object with sunetid parameter;
+   * if already in table, populate other properties.
+   *
+   * @param $su_id
+   */
+  public function __construct($su_id = "") {
+    $su_id = (string) $su_id;
+    $this->sunetid = "";
+    $this->status = self::EARTH_CAPX_INFO_INVALID;
+    $this->etag = "";
+    $this->profilePhotoTimestamp = "";
+    $this->entity_id = 0;
+    $this->profilePhotoFid = 0;
+    if (!empty($su_id)) {
```

Do we want to continue if we get an invalid $su_id? Perhaps this construct should throw an error if the parameter is invalid?

---

I'm inclined to leave this as is. The pre-row-save event already throws a skip exception if there is not SUNet ID and other functions within this object will post migrate messages if the SUNetID is somehow missing without interfering with any upstream processing.

---

Sounds good to me. Perhaps leave a comment inline for the next developer that may review this as well.